### ISSUE TITLE
RunTimeError: Error(s) in loading state_dict for RobertaForSequenceClassification even after transformer==2.9.1

### ISSUE THREAD

**Issue by Marvy101** [openai/gpt-2-output-dataset#35]
I get the following error after trying to run
```
pip install -r requirements.txt
python -m detector.server detector-base.pt
```
Error:
```
RuntimeError: Error(s) in loading state_dict for RobertaForSequenceClassification:
        Missing key(s) in state_dict: "roberta.embeddings.position_ids".
        Unexpected key(s) in state_dict: "roberta.pooler.dense.weight", "roberta.pooler.dense.bias"
```

I have tried ```transformers==2.9.1 ``` in requirements.txt, still receiving the same error.

Thanks for your help.

**Comment by hugoroussel** [comment]
Having the same error too :)

**Comment by Marvy101** [comment]
> Having the same error too :)

Hey, still waiting on a solution and trying different things. 

You wanna have a discussion about AI, and the future?

**Comment by edmundman** [comment]
also having issues with this here is more of the info 

:~/detect$ python -m detector.server detector-large.pt
Loading checkpoint from detector-large.pt
Some weights of the model checkpoint at roberta-large were not used when initializing RobertaForSequenceClassification: ['lm_head.layer_norm.bias', 'lm_head.layer_norm.weight', 'lm_head.dense.bias', 'lm_head.dense.weight', 'roberta.pooler.dense.weight', 'lm_head.bias', 'lm_head.decoder.weight', 'roberta.pooler.dense.bias']
- This IS expected if you are initializing RobertaForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing RobertaForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of RobertaForSequenceClassification were not initialized from the model checkpoint at roberta-large and are newly initialized: ['classifier.out_proj.bias', 'classifier.dense.bias', 'classifier.dense.weight', 'classifier.out_proj.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
Traceback (most recent call last):
  File "/home/ed/mambaforge/envs/detect2/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/ed/mambaforge/envs/detect2/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/ed/detect/detector/server.py", line 120, in <module>
    fire.Fire(main)
  File "/home/ed/mambaforge/envs/detect2/lib/python3.9/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/ed/mambaforge/envs/detect2/lib/python3.9/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/home/ed/mambaforge/envs/detect2/lib/python3.9/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/ed/detect/detector/server.py", line 89, in main
    model.load_state_dict(data['model_state_dict'])
  File "/home/ed/mambaforge/envs/detect2/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1671, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for RobertaForSequenceClassification:
        Missing key(s) in state_dict: "roberta.embeddings.position_ids".
        Unexpected key(s) in state_dict: "roberta.pooler.dense.weight", "roberta.pooler.dense.bias".

**Comment by ps3-app** [comment]
Having the same issues,
`RuntimeError: Error(s) in loading state_dict for RobertaForSequenceClassification:
        Missing key(s) in state_dict: "roberta.embeddings.position_ids".
        Unexpected key(s) in state_dict: "roberta.pooler.dense.weight", "roberta.pooler.dense.bias" `

do you have the solution for it yet?

**Comment by AmirFone** [comment]
I am having the exact same issue.

**Comment by AmirFone** [comment]
was anyone able to resolve this issue yet?

**Comment by hugoroussel** [comment]
No luck on my side :( I think it might be related to #36 (running on macs M1). Also tried running it via docker without luck too.

**Comment by AryanGold** [comment]
I confirm that set `transformers==2.9.1` resolve issue. I try it in isolated Docker container and rebuilding container from scratch, for to be sure that all packages reinstalled.

**Comment by hugoroussel** [comment]
@AryanGold can you give informations about you current setup (OS etc)?

**Comment by AryanGold** [comment]
> 

My Host: Ubuntu 20.04, Docker version 20.10.22

I think important to run  this repository in isolated Docker container or Python Virtual Environment. Also if you change some version in requirements.txt then need reinstall all pip packages without any cache etc. 

I also attache sample of my Docker file.
[Docker_sample.zip](https://github.com/openai/gpt-2-output-dataset/files/10320931/Docker_sample.zip)

**Comment by varenc** [comment]
I also ran into the same error and using `transformers==2.9.1` fixed it. If you're not using a proper enviorment, like we all should be, then you can probably fix the problem just by doing 'pip uninstall transformers` and then `pip install transformers==2.9.1` but WMMV.

**Comment by manueltarouca** [comment]
Make sure that your python version is 3.7.3 and the following package is transformers==2.9.1.

**Comment by SachinGanesh** [comment]
replace 
```
self.model.load_state_dict(data['model_state_dict'])
```
with 
```
self.model.load_state_dict(data['model_state_dict'], strict=False)
```
